backend/auth.ts

```diff
@@ -100,11 +100,15 @@ export async function validateRequest(
 }
 
 export async function requireLogin(
-  req: NextApiRequest,
-  res: NextApiResponse,
+  req: NextApiRequest | IncomingMessage,
+  res: NextApiResponse | ServerResponse,
```

Comment:
Lisäsin `IncomingMessage` ja `ServerResponse` typet tähän, koska `/utils/getServerSideProps.ts`-tiedosto lähettää  `req`:n ja `res`:n noina typeinä

Reply:
Itseasiassa tätä toteutusta ei pystykkään tekemään, koska nyt ei pysty kirjautumaan ollenkaan. Koska jos käyttäjä on kirjautunut ulos, joko `!userDetails.session.isLoggedIn` tai `!userDetails.session` pitää paikkaansa ja näin ollen heittää vaan virhettä